pkgs/top-level/all-packages.nix

```diff
@@ -5534,18 +5534,18 @@ with pkgs;
   # example of an error which this fixes
   # [Errno 8] Exec format error: './gdk3-scan'
   mesonEmulatorHook =
-    if (!stdenv.buildPlatform.canExecute stdenv.targetPlatform) then
+    if (!lib.systems.canExecute stdenv.buildPlatform stdenv.targetPlatform) then
```

Comment:
I added mesonEmulatorHook and it has to be targetPlatform, otherwise it'll always throw. 

AFAIK it's been working well with no issues for about a year

Reply:
> Unrelated to this PR, but why does `targetPlatform` appear here, but nowhere else relating to `mesonEmulatorHook`? Seems to me this should also be `hostPlatform`.

It seems weird to me, but nixpkgs thinks of things like `pkg-config` and `meson` as having a `targetPlatform`.  So `mesonEmulatorHook`'s `targetPlatform` is the `hostPlatform` of the package that lists `mesonEmulatorHook` as a `nativeBuildInput`.

Pretend for a moment that `build.ninja` is a compiler (which itself is "compiled" by `meson`) and it might make sense.

IMHO if a package doesn't emit binaries, it should be built with `stdenv.targetPlatform==null` in order to eliminate an unnecessary degree of freedom.  But getting to that point is a big, long project.

Reply:
Figured out what was irking me, let's discuss at https://github.com/NixOS/nixpkgs/pull/238654.